lib/railway_ipc/core/message_format/json_protobuf.ex

```diff
@@ -152,30 +126,22 @@ defmodule RailwayIpc.Core.MessageFormat.JsonProtobuf do
   defp check_that_module_is_defined({:error, _} = error), do: error
 
   defp parse_type({:ok, %{type: type} = payload}) when is_binary(type) do
-    module = type_to_module(type)
-
-    data_module =
-      if Enum.member?(
-           ["LearnIpc::Events::Student::Registered", "LearnIpc::Events::Student::UpdatedProfile"],
-           type
-         ) do
-        type_to_module("LearnIpc.Entities.Student")
-      else
-        type_to_module("#{type}::Data")
-      end
-
-    {:ok, Map.merge(payload, %{module: module, data_module: data_module})}
+    {:ok, Map.merge(payload, %{module: type_to_module(type)})}
+  end
+
+  defp parse_type({:ok, %{type: type}}) when is_nil(type) do
+    {:error, "Message is missing the `type` attribute"}
   end
 
   defp parse_type({:ok, %{type: type}}) when not is_binary(type) do
     {:error, "Message `type` attribute must be a string"}
   end
 
-  defp parse_type({:ok, _}), do: {:error, "Message is missing the `type` attribute"}
   defp parse_type({:error, _} = error), do: error
 
   defp decode_json(json) do
-    {:ok, Jason.decode!(json, keys: :atoms)}
+    decoded = Jason.decode!(json)
+    {:ok, %{type: decoded["type"], encoded_message: decoded["encoded_message"]}}
```

Comment:
Not that we can't decode the JSON to a map with atom keys, the `Protobuf.JSON.Decode` module expects the map (and any sub-maps) to have string keys.